packages/core/src/loaders/volumeLoader.ts

```diff
@@ -34,34 +34,70 @@ interface LocalVolumeOptions {
   direction: Mat3;
 }
 
-function createInternalVTKRepresentation({
-  dimensions,
-  metadata,
-  spacing,
-  direction,
-  origin,
-  scalarData,
-}): vtkImageDataType {
+/**
+ * Adds a single scalar data to a 3D volume
+ */
+function addScalarDataToImageData(
+  imageData: vtkImageDataType,
+  scalarData: Types.VolumeScalarData,
+  dataArrayAttrs
+) {
+  const scalarArray = vtkDataArray.newInstance({
+    name: `Pixels`,
+    values: scalarData,
+    ...dataArrayAttrs,
+  });
+
+  imageData.getPointData().setScalars(scalarArray);
+}
+
+/**
+ * Adds multiple scalar data (time points) to a 4D volume
+ */
+function addScalarDataArraysToImageData(
+  imageData: vtkImageDataType,
+  scalarDataArrays: Types.VolumeScalarData[],
+  dataArrayAttrs
+) {
+  scalarDataArrays.forEach((scalarData, i) => {
+    const vtkScalarArray = vtkDataArray.newInstance({
+      name: `timePoint-${i}`,
+      values: scalarData,
+      ...dataArrayAttrs,
+    });
+
+    imageData.getPointData().addArray(vtkScalarArray);
+  });
+
+  // Set the first as active otherwise nothing is displayed on the screen
+  imageData.getPointData().setActiveScalars('timePoint-0');
+}
+
+function createInternalVTKRepresentation(volume): vtkImageDataType {
+  const { dimensions, metadata, spacing, direction, origin } = volume;
   const { PhotometricInterpretation } = metadata;
 
   let numComponents = 1;
   if (PhotometricInterpretation === 'RGB') {
     numComponents = 3;
   }
 
-  const scalarArray = vtkDataArray.newInstance({
-    name: 'Pixels',
-    numberOfComponents: numComponents,
-    values: scalarData,
-  });
-
   const imageData = vtkImageData.newInstance();
+  const scalarData = volume.scalarData;
```

Comment:
shoudn't this be `getScalarData` too?

Reply:
Updated since `.scalarData` is now protected